bin/sozo/src/commands/options/account/controller.rs

```diff
@@ -0,0 +1,177 @@
+            trace!(expires_at = %session.expires_at, policies = session.policies.len(), "Found existing session.");
+            session
+        }
+
+        // Return error if user not logged in on slot yet
+        Err(e @ slot::Error::Unauthorized) => {
+            return Err(e.into());
+        }
+
+        // Create a new session if not found or other error
+        Ok(None) | Err(_) => {
+            trace!(%username, chain = format!("{chain_id:#}"), "Creating new session key.");
+            let policies = collect_policies_from_project(contract_address, config)?;
+            let session = slot::session::create(rpc_url, &policies).await?;
+            slot::session::store(chain_id, &session)?;
+            session
+        }
+    };
+
+    let methods = session_details
+        .policies
+        .into_iter()
+        .map(|p| AllowedMethod::new(p.target, &p.method))
+        .collect::<Result<Vec<AllowedMethod>, _>>()?;
+
+    // Copied from `account-wasm` <https://github.com/cartridge-gg/controller/blob/0dd4dd6cbc5fcd3b9a1fd8d63dc127f6312b733f/packages/account-wasm/src/lib.rs#L78-L88>
+    let guardian = SigningKey::from_secret_scalar(short_string!("CARTRIDGE_GUARDIAN"));
+    let signer = SigningKey::from_secret_scalar(session_details.credentials.private_key);
+    // TODO(kariy): make `expires_at` a `u64` type in the session struct
+    let expires_at = session_details.expires_at.parse::<u64>()?;
+    let session = Session::new(methods, expires_at, &signer.signer())?;
+
+    let session_account = SessionAccount::new(
+        provider,
+        signer,
+        guardian,
+        contract_address,
+        chain_id,
+        session_details.credentials.authorization,
+        session,
+    );
+
+    trace!(
+        chain = format!("{chain_id:#}"),
+        address = format!("{contract_address:#x}"),
+        "Created Controller session account"
+    );
+
+    Ok(session_account)
+}
+
+/// Policies are the building block of a session key. It's what defines what methods are allowed for
+/// an external signer to execute using the session key.
+///
+/// This function collect all the contracts' methods in the current project according to the
+/// project's deployment manifest (manifest.toml) and convert them into policies.
+fn collect_policies_from_project(
+    user_address: FieldElement,
+    config: &Config,
+) -> Result<Vec<Policy>> {
+    let root_dir = config.root();
+    let manifest = get_project_deployment_manifest(root_dir, config.profile().as_str())?;
+    let policies = collect_policies(user_address, root_dir, manifest)?;
+    trace!(policies_count = policies.len(), "Extracted policies from project.");
+    Ok(policies)
+}
+
+fn get_project_deployment_manifest(
+    root_dir: &Utf8Path,
+    profile: &str,
+) -> Result<DeploymentManifest> {
+    let mut manifest_path = root_dir.to_path_buf();
+    manifest_path.extend(["manifests", profile, "manifest.toml"]);
+    Ok(DeploymentManifest::load_from_path(&manifest_path)?)
+}
```

Comment:
this would when there is no deployment manifest, and if the user want to deploy with the cartridge account it wouldn't be possible. i think the correct way to do this would be read abi path from `BaseManifest` and calculate the addresses manually to pass into policy.